### PR TITLE
Fix stringification of Vector4

### LINKED ISSUE #99541: Godot4.4-dev5: String presentation of float Vector types has changed and inconsistent
### Tested versions

v4.4.dev5.mono.official [9e6098432]

### System information

all

### Issue description

After upgrade my CI workflow, I run into a lot of test failures according to string formatting changes for float vector types.
I welcome the new format of float vectors, which now also displays floating point notation for integer values
Now Vector2.ONE => "(1.0, 1.0)" before (1, 1)
But it is not consistent over all vector types.

	prints(Vector2(2, 2)) => "(2.0, 2.0)"
	prints(Vector3(2, 2, 2)) => "(2.0, 2.0, 2.0)"
	prints(Vector4(2, 2, 2, 2)) => "(2, 2, 2, 2)"

Vector4 shows the old formatting

### Steps to reproduce

```gd
func print_vectors() -> void:

	prints(Vector2(2, 2))
	prints(Vector3(2, 2, 2))
	prints(Vector4(2, 2, 2, 2))
```
output:
```
(2.0, 2.0)
(2.0, 2.0, 2.0)
(2, 2, 2, 2)
```

### Minimal reproduction project (MRP)

n/a

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #99541
I also improved vector stringification tests.

EDIT:
Also added a fix/test for Projection.

### CODE REVIEW

File: modules/gdscript/tests/scripts/runtime/features/stringify.gd [hunk 1]
```diff
@@ -4,15 +4,18 @@ func test():
...
-	print(Rect2(0.25, 0.25, 0.5, 0.5))
+	print(Rect2(0.25, 0.25, 0.5, 1))
 	print(Rect2i(0, 0, 0, 0))
 
-	print(Vector3(0.25, 0.25, 0.25))
+	print(Vector3(0.25, 0.25, 1))
 	print(Vector3i(0, 0, 0))
 
+	print(Vector4(0.25, 0.25, 0.25, 1))
+	print(Vector4i(0, 0, 0, 0))
```
Comment (akien-mga) [inline modules/gdscript/tests/scripts/runtime/features/stringify.gd:17]: While at it, could you add a print test for `Projection`? It's also missing.

File: core/math/projection.cpp [hunk 2]
```diff
@@ -915,7 +915,12 @@ Projection::operator String() const {
 	String str;
 	for (int i = 0; i < 4; i++) {
 		for (int j = 0; j < 4; j++) {
-			str += String((j > 0) ? ", " : "\n") + rtos(columns[i][j]);
+			if (j > 0) {
+				str += ", ";
+			} else if (i > 0) {
+				str += "\n";
+			}
+			str += rtos(columns[i][j]);
```
Comment (akien-mga) [inline core/math/projection.cpp:923]: Maybe that's material for a separate PR in the end, but I think we should change Projection's string representation to be similar to the other matrices.

So like copied from `Basis` but with `X`, `Y`, `Z`, `W`.

All other Variants print on a single line so this is the odd one out.

Comment (akien-mga) [reply core/math/projection.cpp:923]: So to be clear I'd suggest undoing the changes to add the Projection test here, and make another PR for the above.

Comment (akien-mga) [reply core/math/projection.cpp:923]: I've done this myself in #99563, so you can remove the Projection stuff from this PR.

**Review by akien-mga — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**